README.md

# homebrewanalytics

Access ‘Homebrew’ Formluae Analytics Data

## Description

The ‘Homebrew Project’ \<brew.sh\> has a myriad of “recipes” that make
life easier for ‘macOS’ users by enabling them to (mostly) effortlessly
install popular open source libraries, tools, utilities and
applications. The project collectes anonymous metrics from users who
have not opted-out of metrics collection and makes them available via a
‘JSON’ ‘API’.

## What’s Inside The Tin

The following functions are implemented:

For formulae info:

  - `brew_formulae`: Retrieve formulae metadata for all homebrew-core
    formulae

For info on “events”:

  - `brew_analytics_events`: Retrieve Analytics Events for Homebrew
    Formulae for a Given Category

These are just convenience and legacy API support wrappers for
`brew_analytics_events()`

  - `brew_build_error_events`: Retrieve Analytics Events for Homebrew
    Formulae for a build-error events
  - `brew_cask_install_events`: Retrieve Analytics Events for Homebrew
    Formulae for a cask-install events
  - `brew_install_events`: Retrieve Analytics Events for Homebrew
    Formulae for a install events
  - `brew_install_on_request_events`: Retrieve Analytics Events for
    Homebrew Formulae for install-on-request events
  - `brew_os_version_events`: Retrieve Analytics Events for Homebrew
    Formulae for a OS
events

## Installation

``` r
devtools::install_git("https://git.sr.ht/~hrbrmstr/homebrewanalytics")
# or
devtools::install_gitlab("hrbrmstr/homebrewanalytics")
# or
devtools::install_github("hrbrmstr/homebrewanalytics")
```

## Usage

``` r
library(homebrewanalytics)
library(tibble) # for printing

# current verison
packageVersion("homebrewanalytics")
## [1] '0.2.0'
```

### Formulae info

All:

``` r
brew_formulae()
## # A tibble: 4,694 x 24
##    name  full_name oldname aliases versioned_formu… desc  homepage versions$stable $devel $head $bottle revision
##  * <chr> <chr>     <chr>   <list>  <list>           <chr> <chr>    <chr>           <lgl>  <chr> <lgl>      <int>
##  1 a2ps  a2ps      <NA>    <chr [… <chr [0]>        Any-… https:/… 4.14            NA     <NA>  TRUE           0
##  2 a52d… a52dec    <NA>    <chr [… <chr [0]>        Libr… https:/… 0.7.4           NA     <NA>  TRUE           0
##  3 aacg… aacgain   <NA>    <chr [… <chr [0]>        AAC-… https:/… 1.8             NA     <NA>  TRUE           0
##  4 aalib aalib     <NA>    <chr [… <chr [0]>        Port… https:/… 1.4rc5          NA     <NA>  TRUE           1
##  5 aama… aamath    <NA>    <chr [… <chr [0]>        Rend… http://… 0.3             NA     <NA>  TRUE           0
##  6 aap   aap       <NA>    <chr [… <chr [0]>        Make… http://… 1.094           NA     <NA>  TRUE           0
##  7 aard… aardvark… <NA>    <chr [… <chr [0]>        Util… http://… 1.0             NA     <NA>  TRUE           0
##  8 abcde abcde     <NA>    <chr [… <chr [0]>        Bett… https:/… 2.9.2           NA     HEAD  TRUE           1
##  9 abcl  abcl      <NA>    <chr [… <chr [0]>        Arme… https:/… 1.5.0           NA     HEAD  TRUE           1
## 10 abcm… abcm2ps   <NA>    <chr [… <chr [0]>        ABC … http://… 8.14.2          NA     <NA>  TRUE           0
## # … with 4,684 more rows, and 30 more variables: version_scheme <int>, bottle$stable$rebuild <int>, $$cellar <chr>,
## #   $$prefix <chr>, $$root_url <chr>, $$files$mojave$url <chr>, $$$$sha256 <chr>, $$$high_sierra$url <chr>,
## #   $$$$sha256 <chr>, $$$sierra$url <chr>, $$$$sha256 <chr>, $$$el_capitan$url <chr>, $$$$sha256 <chr>,
## #   $$$yosemite$url <chr>, $$$$sha256 <chr>, $$$mavericks$url <chr>, $$$$sha256 <chr>, keg_only <lgl>, options <list>,
## #   build_dependencies <list>, dependencies <list>, recommended_dependencies <list>, optional_dependencies <list>,
## #   requirements <list>, conflicts_with <list>, caveats <chr>, installed <list>, linked_keg <lgl>, pinned <lgl>,
## #   outdated <lgl>
```

One:

``` r
brew_formulae("curl")
## # A tibble: 1 x 25
##   name  full_name oldname aliases versioned_formu… desc  homepage versions revision version_scheme bottle keg_only
##   <chr> <chr>     <I(lis> <I(lis> <I(list)>        <chr> <chr>    <I(list>    <int>          <int> <I(li> <lgl>   
## 1 curl  curl      <NULL>  <list … <list [0]>       Get … https:/… <list […        0              0 <list… FALSE   
## # … with 13 more variables: options <I(list)>, build_dependencies <chr>, dependencies <I(list)>,
## #   recommended_dependencies <I(list)>, optional_dependencies <I(list)>, requirements <I(list)>,
## #   conflicts_with <I(list)>, caveats <I(list)>, installed <I(list)>, linked_keg <I(list)>, pinned <lgl>,
## #   outdated <lgl>, analytics <I(list)>
```

Two:

``` r
brew_formulae(c("curl", "wget"))
## # A tibble: 2 x 25
##   name  full_name oldname aliases versioned_formu… desc  homepage versions revision version_scheme bottle keg_only
##   <chr> <chr>     <I(lis> <I(lis> <I(list)>        <chr> <chr>    <I(list>    <int>          <int> <I(li> <lgl>   
## 1 curl  curl      <NULL>  <list … <list [0]>       Get … https:/… <list […        0              0 <list… FALSE   
## 2 wget  wget      <NULL>  <list … <list [0]>       Inte… https:/… <list […        3              0 <list… FALSE   
## # … with 13 more variables: options <I(list)>, build_dependencies <chr>, dependencies <I(list)>,
## #   recommended_dependencies <I(list)>, optional_dependencies <I(list)>, requirements <I(list)>,
## #   conflicts_with <I(list)>, caveats <I(list)>, installed <I(list)>, linked_keg <I(list)>, pinned <lgl>,
## #   outdated <lgl>, analytics <I(list)>
```

Errant name but still retrieves the rest:

``` r
brew_formulae(c("curl", "wget", "doesnotexist"))
## # A tibble: 2 x 25
##   name  full_name oldname aliases versioned_formu… desc  homepage versions revision version_scheme bottle keg_only
##   <chr> <chr>     <I(lis> <I(lis> <I(list)>        <chr> <chr>    <I(list>    <int>          <int> <I(li> <lgl>   
## 1 curl  curl      <NULL>  <list … <list [0]>       Get … https:/… <list […        0              0 <list… FALSE   
## 2 wget  wget      <NULL>  <list … <list [0]>       Inte… https:/… <list […        3              0 <list… FALSE   
## # … with 13 more variables: options <I(list)>, build_dependencies <chr>, dependencies <I(list)>,
## #   recommended_dependencies <I(list)>, optional_dependencies <I(list)>, requirements <I(list)>,
## #   conflicts_with <I(list)>, caveats <I(list)>, installed <I(list)>, linked_keg <I(list)>, pinned <lgl>,
## #   outdated <lgl>, analytics <I(list)>
```

### Analytic Events

Retrieve formlua install events for the past 30, 90 or 365 days. We’ll
juse use the default:

``` r
brew_install_events()
## # A tibble: 10,000 x 9
##    number formula   count percent category start_date end_date   total_items total_count
##  *  <int> <chr>     <dbl>   <dbl> <chr>    <date>     <date>           <int>       <int>
##  1      1 readline 502756  0.0306 install  2019-01-03 2019-02-02       14585    16417871
##  2      2 openssl  495728  0.0302 install  2019-01-03 2019-02-02       14585    16417871
##  3      3 python   478566  0.0291 install  2019-01-03 2019-02-02       14585    16417871
##  4      4 sqlite   478056  0.0291 install  2019-01-03 2019-02-02       14585    16417871
##  5      5 node     347418  0.0212 install  2019-01-03 2019-02-02       14585    16417871
##  6      6 icu4c    296464  0.0181 install  2019-01-03 2019-02-02       14585    16417871
##  7      7 gdbm     289269  0.0176 install  2019-01-03 2019-02-02       14585    16417871
##  8      8 xz       234537  0.0143 install  2019-01-03 2019-02-02       14585    16417871
##  9      9 libpng   219181  0.0134 install  2019-01-03 2019-02-02       14585    16417871
## 10     10 glib     196391  0.012  install  2019-01-03 2019-02-02       14585    16417871
## # … with 9,990 more rows
```

### Install On Request Events

Retrieve formlua install-on-request events for the past 30, 90 or 365
days

``` r
brew_install_on_request_events()
## # A tibble: 10,000 x 9
##    number formula      count percent category           start_date end_date   total_items total_count
##  *  <int> <chr>        <dbl>   <dbl> <chr>              <date>     <date>           <int>       <int>
##  1      1 python      264769  0.0451 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  2      2 node        260559  0.0444 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  3      3 wget        166510  0.0284 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  4      4 git         119313  0.0203 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  5      5 yarn        114583  0.0195 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  6      6 youtube-dl   87835  0.015  install_on_request 2019-01-03 2019-02-02       13747     5872029
##  7      7 imagemagick  83740  0.0143 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  8      8 vim          78656  0.0134 install_on_request 2019-01-03 2019-02-02       13747     5872029
##  9      9 cmake        75337  0.0128 install_on_request 2019-01-03 2019-02-02       13747     5872029
## 10     10 postgresql   74322  0.0127 install_on_request 2019-01-03 2019-02-02       13747     5872029
## # … with 9,990 more rows
```

### Build Error Events Events

Retrieve formlua build error events for the past 30

``` r
brew_build_error_events()
## # A tibble: 2,476 x 9
##    number formula              count percent category   start_date end_date   total_items total_count
##  *  <int> <chr>                <dbl>   <dbl> <chr>      <date>     <date>           <int>       <int>
##  1      1 libimobiledevice     22705  0.341  BuildError 2019-01-03 2019-02-02        2476       66646
##  2      2 qt                    3846  0.0577 BuildError 2019-01-03 2019-02-02        2476       66646
##  3      3 node                  1699  0.0255 BuildError 2019-01-03 2019-02-02        2476       66646
##  4      4 go                    1540  0.0231 BuildError 2019-01-03 2019-02-02        2476       66646
##  5      5 graphite2             1526  0.0229 BuildError 2019-01-03 2019-02-02        2476       66646
##  6      6 valgrind              1515  0.0227 BuildError 2019-01-03 2019-02-02        2476       66646
##  7      7 usbmuxd               1296  0.0194 BuildError 2019-01-03 2019-02-02        2476       66646
##  8      8 python                1224  0.0184 BuildError 2019-01-03 2019-02-02        2476       66646
##  9      9 openssl               1217  0.0183 BuildError 2019-01-03 2019-02-02        2476       66646
## 10     10 facebook/fb/fbsimctl  1036  0.0155 BuildError 2019-01-03 2019-02-02        2476       66646
## # … with 2,466 more rows
```

``` r
cloc::cloc_pkg_md()
```

| Lang | \# Files |  (%) | LoC | (%) | Blank lines |  (%) | \# Lines |  (%) |
| :--- | -------: | ---: | --: | --: | ----------: | ---: | -------: | ---: |
| R    |        6 | 0.86 | 143 | 0.9 |          49 | 0.56 |       84 | 0.58 |
| Rmd  |        1 | 0.14 |  16 | 0.1 |          39 | 0.44 |       61 | 0.42 |

## Code of Conduct

Please note that the ‘homebrewanalytics’ project is released with a
[Contributor Code of Conduct](CODE_OF_CONDUCT.md). By contributing to
this project, you agree to abide by its terms.


NEWS.md
0.2.0
* Updated to the new "API"
* Added tests
* Updated documentation

0.1.0 
* Initial release


tests/test-all.R
library(testthat)
test_check("homebrewanalytics")


tests/testthat/test-homebrewanalytics.R
context("api calls work as expected")
test_that("we can do something", {

  expect_is(brew_formulae(), "data.frame")
  expect_is(brew_formulae("curl"), "data.frame")
  expect_is(brew_formulae(c("curl", "wget")), "data.frame")
  expect_warning(brew_formulae("nothere"))
  expect_null(brew_formulae("nothere"))

  expect_is(brew_build_error_events(), "data.frame")
  expect_is(brew_cask_install_events(), "data.frame")
  expect_is(brew_install_events(), "data.frame")
  expect_is(brew_install_on_request_events(), "data.frame")
  expect_is(brew_os_version_events(), "data.frame")

})


R/build-error.R
#' Retrieve Build Errors for the past 30 days for Homebrew Formulae
#'
#' @md
#' @return data frame (tibble) with build recipe and count;
#'         also includes a `meta` attribute with the metrics category and
#'         metrics tart/end date.
#' @export
#' @references <https://formulae.brew.sh/analytics/>
#' @examples
#' xdf <- brew_build_error_events()
#' attr(xdf, "meta")
brew_build_error_events <- function() {

  httr::GET(
    url = "https://formulae.brew.sh/api/analytics/build-error/homebrew-core/30d.json",
    httr::user_agent(HBUA)
  ) -> res

  httr::stop_for_status(res)

  out <- httr::content(res, as = "text", encoding = "UTF-8")
  out <- jsonlite::fromJSON(out)

  meta <- out[c("category", "start_date", "end_date")]
  meta$start_date <- as.Date(meta$start_date)
  meta$end_date <- as.Date(meta$end_date)

  do.call(
    rbind.data.frame,
    lapply(out$formulae, function(.x) {
      .x$count <- as.numeric(gsub(",", "", .x$count))
      .x <- .x[,c("formula", "count")]
      colnames(.x) <- c("build_recipe", "count")
      .x
    })
  ) -> out

  class(out) <- c("tbl_df", "tbl", "data.frame")

  attr(out, "meta") <- meta

  out

}


R/brew-formulae.R
..one_formula <- function(formula) {

  # for testig
  # formula <- "wget"

  httr::GET(
    url = sprintf(
      "https://formulae.brew.sh/api/formula/%s.json", formula
    ), .HBUA
  ) -> res

  if (httr::status_code(res) != 200) {
    warning(
      "'", formula, "' not found in Homebrew/homebrew-core. Skipping...",
      call.=FALSE
    )
    return(NULL)
  }

  f <- httr::content(res, as = "text")
  f <- jsonlite::fromJSON(f)

  which(
    vapply(f, is_length_zero, FUN.VALUE = logical(1), USE.NAMES = FALSE)
  ) -> idx1

  which(
    vapply(f, is_length_gt_one, FUN.VALUE = logical(1), USE.NAMES = FALSE)
  ) -> idx2

  which(
    vapply(f, class, FUN.VALUE = character(1), USE.NAMES = FALSE) == "list"
  ) -> idx3

  idx <- unique(c(idx1, idx2, idx3))

  for (i in idx) {
    f[[i]] <- I(list(f[[i]]))
  }

  out <- data.frame(f, stringsAsFactors = FALSE)

  class(out) <- c("tbl_df", "tbl", "data.frame")

  out

}

.one_formula <- memoise::memoise(..one_formula)

..all_formulae <- function() {

  httr::GET(
    url = "https://formulae.brew.sh/api/formula.json", .HBUA
  ) -> res

  httr::stop_for_status(res)
  out <- httr::content(res, as = "text")
  out <- jsonlite::fromJSON(out)

  class(out) <- c("tbl_df", "tbl", "data.frame")
  out

}

.all_formulae <- memoise::memoise(..all_formulae)

.brew_formulae <- function(formulae = "all") {

  if ((length(formulae) == 1) && (formulae == "all")) {
    return(.all_formulae())
  }

  if ((length(formulae) == 1) && (formulae != "all")) {
    return(.one_formula(formulae))
  }

  if (length(formulae) > 1) {

    if ("all" %in% formulae) {
      stop(
        "Cannot combine 'all' witt a specified vector of formulae names.",
        call.=FALSE
      )
    }

    f <- lapply(formulae, .one_formula)
    bad <- vapply(f, is.null, FUN.VALUE = logical(1), USE.NAMES = FALSE)
    f <- f[!bad]

    if (length(f) > 0) {
      out <- do.call(rbind.data.frame, f)
      class(out) <- c("tbl_df", "tbl", "data.frame")
      return(out)
    } else {
      message("None of the specified formulae found.")
    }

  }

}

#' Retrieve formulae metadata for all homebrew-core formulae
#'
#' Get the `brew info --json=v1`` output for *all* current
#' Homebrew/homebrew-core formulae _or_ get the same output for
#' one or more current Homebrew/homebrew-core formulae with an extra
#' analytics key with analytics data by specify
#'
#' @md
#' @param formluae if "`all`" (the default) then metadata on all
#'        Homebrew/homebrew-core formulae will be returned. Otherwise
#'        the API will be queried for all formulae in the passed in
#'        character vector and returned, skipping (and warning) on
#'        any specified formula that does not exist.
#' @return data frame (tibble) with formula analytics data or `NULL` if
#'         what you looked for was not found.
#' @note The results of the API call are memoised as they won't likely
#'       won't change between calls in the same R session. Use the
#'       facilities provided by the `memoise` package to clear the
#'       cache if you use this in a long running R session.
#' @export
#' @references <https://formulae.brew.sh/docs/api/>
#' @examples
#' f <- brew_formulae() # same as manually specifying "all"
#' f <- brew_formulae("curl")
#' f <- brew_formulae(c("curl", "youtube-dl", "blemenge"))
brew_formulae <- memoise::memoise(.brew_formulae)


R/install-events.R
#' Retrieve Install Events for Homebrew Formulae
#'
#' @md
#' @param delta number of previous days to return. One of `30`, `90`, `365`
#' @return data frame (tibble) with formula, configure options and count;
#'         also includes a `meta` attribute with the metrics category and
#'         metrics tart/end date.
#' @export
#' @references <https://formulae.brew.sh/analytics/>
#' @examples
#' xdf <- brew_install_events(30)
#' attr(xdf, "meta")
brew_install_events <- function(delta = c(30, 90, 365)) {

  stopifnot(delta[1] %in% c(30, 90, 365))

  httr::GET(
    url = sprintf(
      "https://formulae.brew.sh/api/analytics/install/homebrew-core/%sd.json",
      as.integer(delta[1])
    ),
    httr::user_agent(HBUA)
  ) -> res

  httr::stop_for_status(res)

  out <- httr::content(res, as = "text", encoding = "UTF-8")
  out <- jsonlite::fromJSON(out)

  meta <- out[c("category", "start_date", "end_date")]
  meta$start_date <- as.Date(meta$start_date)
  meta$end_date <- as.Date(meta$end_date)

  do.call(
    rbind.data.frame,
    lapply(out$formulae, function(.x) {
      if (nrow(.x) > 1) {
        .x$opts <- trimws(sub(sprintf("%s", .x$formula[1]), "", .x$formula, fixed=TRUE))
        .x$formula <- .x$formula[1]
      }
      .x$opts[1] <- NA_character_
      .x$count <- as.numeric(gsub(",", "", .x$count))
      .x[,c("formula", "opts", "count")]
    })
  ) -> out

  class(out) <- c("tbl_df", "tbl", "data.frame")

  attr(out, "meta") <- meta

  out

}


R/install-on-request-events.R
#' Retrieve Install-on-Requests nts for Homebrew Formulae
#'
#' @md
#' @param delta number of previous days to return. One of `30`, `90`, `365`
#' @return data frame (tibble) with formula, configure options and count;
#'         also includes a `meta` attribute with the metrics category and
#'         metrics tart/end date.
#' @export
#' @references <https://formulae.brew.sh/analytics/>
#' @examples
#' xdf <- brew_install_on_request_events(30)
#' attr(xdf, "meta")
brew_install_on_request_events <- function(delta = c(30, 90, 365)) {

  stopifnot(delta[1] %in% c(30, 90, 365))

  httr::GET(
    url = sprintf(
      "https://formulae.brew.sh/api/analytics/install-on-request/homebrew-core/%sd.json",
      as.integer(delta[1])
    ),
    httr::user_agent(HBUA)
  ) -> res

  httr::stop_for_status(res)

  out <- httr::content(res, as = "text", encoding = "UTF-8")
  out <- jsonlite::fromJSON(out)

  meta <- out[c("category", "start_date", "end_date")]
  meta$start_date <- as.Date(meta$start_date)
  meta$end_date <- as.Date(meta$end_date)

  do.call(
    rbind.data.frame,
    lapply(out$formulae, function(.x) {
      if (nrow(.x) > 1) {
        .x$opts <- trimws(sub(sprintf("%s", .x$formula[1]), "", .x$formula, fixed=TRUE))
        .x$formula <- .x$formula[1]
      }
      .x$opts[1] <- NA_character_
      .x$count <- as.numeric(gsub(",", "", .x$count))
      .x[,c("formula", "opts", "count")]
    })
  ) -> out

  class(out) <- c("tbl_df", "tbl", "data.frame")

  attr(out, "meta") <- meta

  out

}


R/analytic-events.R
.brew_analytics_events <- function(category = c("install",
                                                "install-on-request",
                                                "cask-install",
                                                "build-error",
                                                "os-version"),
                                   delta = c(30, 90, 365)) {

  category <- match.arg(
    category[1],
    c("install", "install-on-request", "cask-install", "build-error", "os-version")
  )

  stopifnot(delta[1] %in% c(30, 90, 365))

  httr::GET(
    url = sprintf(
      "https://formulae.brew.sh/api/analytics/%s/%sd.json",
      category,
      as.integer(delta[1])
    ),
    .HBUA
  ) -> res

  httr::stop_for_status(res)

  out <- httr::content(res, as = "text", encoding = "UTF-8")
  out <- jsonlite::fromJSON(out)

  items <- out[["items"]]
  items[["category"]] <- out[["category"]]
  items[["start_date"]] <- as.Date(out[["start_date"]])
  items[["end_date"]] <- as.Date(out[["end_date"]])
  items[["total_items"]] <- out[["total_items"]]
  items[["total_count"]] <- out[["total_count"]]
  items[["percent"]] <- as.numeric(items[["percent"]]) / 100
  items[["count"]] <- as.numeric(gsub("[^[:digit:]]", "", items[["count"]]))

  class(items) <- c("tbl_df", "tbl", "data.frame")

  items

}

#' Retrieve Analytics Events for Homebrew Formulae for a Given Category
#'
#' Get analytics event data such as event count, percent and time range.
#'
#' The old Homebrew Analytics API had different endpoints for analytics
#' event metrics. These have been kept as wrappers to the new function and
#' other conveneince functions have been added. Thse are:
#'
#' - `[brew_build_error_events()]`
#' - `[brew_cask_install_events()]`
#' - `[brew_install_events()]`
#' - `[brew_install_on_request_events()]`
#' - `[brew_os_version_events()]`
#'
#' @md
#' @param category the category of the analytics events (the default is `install` events.
#'    The following are acceptable inputs:
#'    - `install`: the installation of all formulae
#'    - `install-on-request`: the requested installation of all formulae (i.e. not as a dependency of other formulae)
#'    - `cask-install`: the installation of all casks
#'    - `build-error`: the installation failure of all formulae
#'    - `os-version`: the macOS version of the machine that submitted an event
#' @param delta number of previous days to return. One of `30`, `90`, `365`
#' @note The results of the API call are memoised as they won't likely
#'       won't change between calls in the same R session. Use the
#'       facilities provided by the `memoise` package to clear the
#'       cache if you use this in a long running R session.
#' @return data frame (tibble)
#' @export
#' @references <https://formulae.brew.sh/docs/api/>
#' @examples
#' xdf <- brew_analytics_events("install", 30)
brew_analytics_events <- memoise::memoise(.brew_analytics_events)

#' @rdname brew_analytics_events
#' @export
brew_install_events <- function(delta = c(30, 90, 365)) {
  brew_analytics_events("install", delta)
}

#' @rdname brew_analytics_events
#' @export
brew_install_on_request_events <- function(delta = c(30, 90, 365)) {
  brew_analytics_events("install-on-request", delta)
}

#' @rdname brew_analytics_events
#' @export
brew_build_error_events <- function(delta = c(30, 90, 365)) {
  brew_analytics_events("build-error", delta)
}

#' @rdname brew_analytics_events
#' @export
brew_cask_install_events <- function(delta = c(30, 90, 365)) {
  brew_analytics_events("cask-install", delta)
}

#' @rdname brew_analytics_events
#' @export
brew_os_version_events <- function(delta = c(30, 90, 365)) {
  brew_analytics_events("os-version", delta)
}
